superset/views/core.py

```diff
@@ -1421,6 +1421,48 @@ def fave_slices(  # pylint: disable=no-self-use
             payload.append(dash)
         return json_success(json.dumps(payload, default=utils.json_int_dttm_ser))
 
+    @api
+    @has_access_api
+    @event_logger.log_this
+    @expose("/dashboard/<dashboard_id>/stop/", methods=["POST"])
+    def stop_dashboard_queries(  # pylint: disable=no-self-use
+        self, dashboard_id: int
+    ) -> FlaskResponse:
+        if is_feature_enabled("STOP_DASHBOARD_PENDING_QUERIES"):
+            username = g.user.username if g.user else None
+            dashboard = db.session.query(Dashboard).filter_by(id=dashboard_id).one()
+            slices = dashboard.slices
+            datasource_ids = set()
+            database_ids = set()
+
+            # find databases for all charts in a given dashboard
+            # stop pending query is only available for certain database(s)
+            for slc in slices:
```

Comment:
this is a good candidate for the util function - get database ids from the dashboard

by the way it would be much more efficient if that logic could be done in the SQL itself

Reply:
moved to util function, and added unit test.